relayer_api/services/nodeConnection.js

```diff
@@ -0,0 +1,105 @@
+let url = require('../config/settings').server.url;
+let event = require('../config/events')
+let ioClient = require('socket.io-client');
+let connection = process.env.connection;
+let connectedNodesList = [];
+let allSockets = [];
+let socektsUrl = new Map();
+
+const databaseManager = require('./database/databaseManager')
+
+
+module.exports = {
+    init: function(io) {
+        // Incoming
+        io.on(event.connection, function (socket) {
+            allSockets.push(socket)
+            socket.on(event.incommingConnection, function(connectedUrl){
+                connectedNodesList.push(connectedUrl)
+                socektsUrl.set(socket, connectedUrl);
+            })
+            socket.emit(event.nodesList, connectedNodesList);
+            setEventHandlers(socket);
+        });
+        if(!connection){
+            return;
+        }
+        connectToNodes([connection])      
+    },
+
+
+    emitIdDataOnCreation: function(data){
```

Comment:
I feel that this can very easily be confused for actual deployment as opposed to Counterfactual creation. Maybe we should rename this to `emitCounterfactualIdentityData`